### PR TITLE
Redesign Cards fixes

### PR DESCRIPTION
**Pull request by alexmylonas**
## Background

Fixes based on feedback on #2051 

## Testing

- Locally visual

### CODE REVIEW

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 1]
```diff
@@ -59,14 +64,22 @@ const GenericItemCard: React.FC & GenericItemCardComposition = ({ children }) =>
...
-    <Flex justify="space-between" fontWeight="medium" width={0.7} wordBreak="break-word" {...rest}>
+    <Flex as="header" align="start" mb={2}>
       {children}
     </Flex>
   );
 };
 
+const GenericItemCardHeading: React.FC<TextProps> = ({ children, ...rest }) => {
+  return (
+    <Text as="h4" fontWeight="medium" mr="auto" maxWidth="70%" wordBreak="break-word" {...rest}>
```
Comment (alexmylonas) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:77]: The maxWidth="70%" is based on designs
From mocks: Screenshot 2020-11-19 at 5 39 06 PM

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 2]
```diff
@@ -87,12 +100,12 @@ const GenericItemCardLogo: React.FC<GenericItemCardLogoProps> = ({ src }) => {
...
 };
 
-const GenericItemCardDropdown = React.forwardRef<HTMLButtonElement>(function GenericItemCardOptions(
+const GenericItemCardOptionsButton = React.forwardRef<HTMLButtonElement>(function OptionsButton(
   props,
   ref
 ) {
   return (
-    <Box mt={-1}>
+    <Box ml={1} mt={-1}>
```
Comment (alexmylonas) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:108]: Those margins are need to align with the date text that sometimes we render before options: 
Screenshot 2020-11-19 at 5 40 48 PM

File: web/src/components/LimitItemDisplay/LimitItemDisplay.tsx [hunk 3]
```diff
@@ -51,19 +52,9 @@ const LimitItemDisplay: React.FC<LimitItemDisplayProps> = ({ limit, children })
...
-          borderRadius="circle"
-          fontSize="2x-small"
-          fontWeight="medium"
-          cursor="default"
-        >
-          +{childrenCount - limit}
-        </Flex>
+        <Box>
+          <GenericItemCard.ExtraBadge badge={`+${childrenCount - limit}`} />
+        </Box>
```
Comment (alexmylonas) [inline web/src/components/LimitItemDisplay/LimitItemDisplay.tsx:57]: Added `GenericItemCard.ExtraBadge` here

Without `<Box>` the tooltip content wasn't rendering :/

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 4]
```diff
@@ -108,20 +121,12 @@ const GenericItemCardDropdown = React.forwardRef<HTMLButtonElement>(function Gen
 
 const GenericItemCardDate: React.FC<GenericItemCardDate> = ({ date, ...rest }) => {
   return (
-    <Text fontSize="small" as="span" color="gray-500" {...rest}>
+    <Text fontSize="small" as="time" color="gray-500" {...rest}>
```
Comment (alexmylonas) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:124]: Changed this to `time` we dont always render time, there could some extra text. Like on `LogSourceCard`: 
Screenshot 2020-11-19 at 5 46 44 PM

File: web/src/components/GenericItemCard/GenericItemCard.tsx [hunk 5]
```diff
@@ -59,14 +64,22 @@ const GenericItemCard: React.FC & GenericItemCardComposition = ({ children }) =>
   );
 };
 
-const GenericItemCardHeading: React.FC<TextProps> = ({ children, ...rest }) => {
+const GenericItemCardHeader: React.FC = ({ children }) => {
   return (
-    <Flex justify="space-between" fontWeight="medium" width={0.7} wordBreak="break-word" {...rest}>
+    <Flex as="header" align="start" mb={2}>
```
Comment (3nvi) [inline web/src/components/GenericItemCard/GenericItemCard.tsx:69]: `start` is not a valid value I believe

```suggestion
    <Flex as="header" align="flex-start" mb={2}>
```

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 1 above)
Comment (3nvi) [reply web/src/components/GenericItemCard/GenericItemCard.tsx:77]: I feel there's no reason to restrict it (especially given responsive screens), but I'll leave it to you to decide 😸

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 2 above)
Comment (3nvi) [reply web/src/components/GenericItemCard/GenericItemCard.tsx:108]: do we really need `ml={-1}`? that pushes it "towards the left". Don't we need the opposite?

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 4 above)
Comment (3nvi) [reply web/src/components/GenericItemCard/GenericItemCard.tsx:124]: hmm yeah you are right. Your call! Change it or leave it

File: web/src/components/LimitItemDisplay/LimitItemDisplay.tsx (hunk 3 above)
Comment (3nvi) [reply web/src/components/LimitItemDisplay/LimitItemDisplay.tsx:57]: Not sure on this one. This component doesn't relate to `GenericItemCard` in any way. I do understand the idea of keeping a component for the badge though. How would you feel if we renamed it to `NotificationJewel` (or smth) and we added it as a simple component under `/components`?

**Review by 3nvi — APPROVED** [review]
minor comments, but looks great overall

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 5 above)
Comment (alexmylonas) [reply web/src/components/GenericItemCard/GenericItemCard.tsx]: Done!

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 2 above)
Comment (alexmylonas) [reply web/src/components/GenericItemCard/GenericItemCard.tsx:107]: `ml=[1]` just to provide a bit space between this and date text

File: web/src/components/GenericItemCard/GenericItemCard.tsx (hunk 4 above)
Comment (alexmylonas) [reply web/src/components/GenericItemCard/GenericItemCard.tsx]: Changed!

File: web/src/components/LimitItemDisplay/LimitItemDisplay.tsx (hunk 3 above)
Comment (alexmylonas) [reply web/src/components/LimitItemDisplay/LimitItemDisplay.tsx:57]: Added this as `NotificationJewel`

**Pull request merged**